src/common/NetworkEventManager.ts

```diff
@@ -0,0 +1,101 @@
+import { Protocol } from 'devtools-protocol';
+import { HTTPRequest } from './HTTPRequest.js';
+
+export type QueuedEvents = {
+  responseReceived: Protocol.Network.ResponseReceivedEvent;
+  promise: Promise<void>;
+  resolver: () => void;
+  loadingFinished?: Protocol.Network.LoadingFinishedEvent;
+  loadingFailed?: Protocol.Network.LoadingFailedEvent;
+};
+
+export type RedirectInfoMap = Array<{
+  event: Protocol.Network.RequestWillBeSentEvent;
+  fetchRequestId?: string;
```

Comment:
Let's define local types:

```
type FetchRequestId = string;
type NetworkRequestId = string;
```

and use it for documentation purposes in the maps and methods in the class.

Reply:
🤯 nice idea! Done.